typings/styled-components.d.ts

```diff
@@ -1,122 +1,208 @@
+  ): FlattenInterpolation<ThemedStyledProps<P, T>>[]
 }
 
 // Helper type operators
-type KeyofBase = keyof any;
-type Diff<T extends KeyofBase, U extends KeyofBase> = ({ [P in T]: P } & { [P in U]: never })[T];
-type Omit<T, K extends keyof T> = Pick<T, Diff<keyof T, K>>;
-type DiffBetween<T, U> = Pick<T, Diff<keyof T, keyof U>> & Pick<U, Diff<keyof U, keyof T>>;
-type WithOptionalTheme<P extends { theme?: T; }, T> = Omit<P, "theme"> & { theme?: T; };
+type KeyofBase = keyof any
+type Diff<T extends KeyofBase, U extends KeyofBase> = ({ [P in T]: P } &
+  { [P in U]: never })[T]
+type Omit<T, K extends keyof T> = Pick<T, Diff<keyof T, K>>
+type DiffBetween<T, U> = Pick<T, Diff<keyof T, keyof U>> &
+  Pick<U, Diff<keyof U, keyof T>>
+type WithOptionalTheme<P extends { theme?: T }, T> = Omit<P, 'theme'> & {
+  theme?: T
+}
 
 export interface ThemedStyledComponentsModule<T> {
-  default: ThemedStyledInterface<T>;
+  default: ThemedStyledInterface<T>
+
+  css: ThemedCssFunction<T>
+  keyframes(
+    strings: TemplateStringsArray,
+    ...interpolations: SimpleInterpolation[]
+  ): string
+  injectGlobal(
+    strings: TemplateStringsArray,
+    ...interpolations: SimpleInterpolation[]
+  ): void
+  withTheme<P extends { theme?: T }>(
+    component: React.ComponentType<P>
+  ): ComponentClass<WithOptionalTheme<P, T>>
+
+  ThemeProvider: ThemeProviderComponent<T>
+}
 
-  css: ThemedCssFunction<T>;
-  keyframes(strings: TemplateStringsArray, ...interpolations: SimpleInterpolation[]): string;
-  injectGlobal(strings: TemplateStringsArray, ...interpolations: SimpleInterpolation[]): void;
-  withTheme<P extends { theme?: T; }>(component: React.ComponentType<P>): ComponentClass<WithOptionalTheme<P, T>>;
+declare const styled: StyledInterface
 
-  ThemeProvider: ThemeProviderComponent<T>;
-}
+export const css: ThemedCssFunction<any>
+
+export function withTheme<P extends { theme?: T }, T>(
+  component: React.ComponentType<P>
+): ComponentClass<WithOptionalTheme<P, T>>
 
-declare const styled: StyledInterface;
+export function keyframes(
+  strings: TemplateStringsArray,
+  ...interpolations: SimpleInterpolation[]
+): string
 
-export const css: ThemedCssFunction<any>;
-export function withTheme<P extends { theme?: T; }, T>(component: React.ComponentType<P>): ComponentClass<WithOptionalTheme<P, T>>;
+export function injectGlobal(
+  strings: TemplateStringsArray,
+  ...interpolations: SimpleInterpolation[]
+): void
 
-export function keyframes(strings: TemplateStringsArray, ...interpolations: SimpleInterpolation[]): string;
-export function injectGlobal(strings: TemplateStringsArray, ...interpolations: SimpleInterpolation[]): void;
-export function consolidateStreamedStyles(): void;
-export function isStyledComponent(target: string | Component<object>): boolean;
+export function consolidateStreamedStyles(): void
 
-export const ThemeProvider: ThemeProviderComponent<object>;
+export function isStyledComponent(target: any): target is StyledComponentClass
```

Comment:
I got error on this line:

```
ERROR in [at-loader] ./node_modules/styled-components/typings/styled-components.d.ts:179:59    
TS2707: Generic type 'StyledComponentClass<P, T, O>' requires between 2 and 3 type arguments.
```

Using with awesome-typescript-loader and typescript 3.0.1

Reply:
I am using typescript 2.9.2 and getting the same error.

Reply:
Same here.

Reply:
Is fixed in PR #1885

Reply:
@PzYon But not present in the published version @2.4.1 on the npm :(
/cc @probablyup